applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -406,6 +406,104 @@ static bool sink_parse_cb(struct bt_data *data, void *user_data)
 	return true;
 }
 
+static void set_color_if_supported(char *str, uint16_t bitfield, uint16_t mask)
+{
+	if (bitfield & mask) {
+		strcat(str, COLOR_GREEN);
+	} else {
+		strcat(str, COLOR_RED);
+	}
+}
+
+static bool cap_print_cb(struct bt_data *data, void *user_data)
+{
+	if (data->type == BT_AUDIO_CODEC_CAP_TYPE_FREQ) {
+		uint16_t freq_bit = sys_get_le16(data->data);
+		char supported_freq[320] = "";
+
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_8KHZ);
+		strcat(supported_freq, "8, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_11KHZ);
+		strcat(supported_freq, "11.025, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_16KHZ);
+		strcat(supported_freq, "16, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_22KHZ);
+		strcat(supported_freq, "22.05, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_24KHZ);
+		strcat(supported_freq, "24, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_32KHZ);
+		strcat(supported_freq, "32, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_44KHZ);
+		strcat(supported_freq, "44.1, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_48KHZ);
+		strcat(supported_freq, "48, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_88KHZ);
+		strcat(supported_freq, "88.2, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_96KHZ);
+		strcat(supported_freq, "96, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_176KHZ);
+		strcat(supported_freq, "176, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_192KHZ);
+		strcat(supported_freq, "192, ");
+		set_color_if_supported(supported_freq, freq_bit, BT_AUDIO_CODEC_CAP_FREQ_384KHZ);
+		strcat(supported_freq, "384");
+
+		LOG_INF("\tFrequencies kHz: %s", supported_freq);
+	}
+
+	if (data->type == BT_AUDIO_CODEC_CAP_TYPE_DURATION) {
+		uint16_t dur_bit = sys_get_le16(data->data);
+		char supported_dur[30] = "";
+
+		set_color_if_supported(supported_dur, dur_bit, BT_AUDIO_CODEC_CAP_DURATION_7_5);
+		strcat(supported_dur, "7.5, ");
+		set_color_if_supported(supported_dur, dur_bit, BT_AUDIO_CODEC_CAP_DURATION_10);
+		strcat(supported_dur, "10");
+
+		LOG_INF("\tFrame duration ms: %s", supported_dur);
+	}
+
+	if (data->type == BT_AUDIO_CODEC_CAP_TYPE_CHAN_COUNT) {
+		uint16_t chan_bit = sys_get_le16(data->data);
+		char supported_chan[120] = "";
+
+		set_color_if_supported(supported_chan, chan_bit, BT_AUDIO_CODEC_CAP_CHAN_COUNT_1);
```

Comment:
```suggestion
		for (int i = 0; i < 8; i++) {
		        char tmp_str[4];
		        set_color_if_supported(supported_chan, chan_bit, BIT(i);
		        sprintf(tmp_str, "%d, ", i);
		        strcat(supported_chan, tmp_str);
		}
```

Reply:
Discussed offline, this is going from an Enum to just hardcoded bits, if anything changes in the spec/host it will not changed here, causing a bug